validphys2/src/validphys/inconsistent_ct.py

```diff
@@ -0,0 +1,142 @@
+import dataclasses
+from validphys.coredata import CommonData
+import pandas as pd
+
+
+
+@dataclasses.dataclass(eq = False)
+class InconsistentCommonData(CommonData):
+    """
+    Class that inherits all of the methods
+    of coredata.CommonData class.
+    
+    This class is meant to have all the 
+    methods needed in order to introduce 
+    an inconsistency within a Closure Test.
+    """
+
+    setname: str
+    ndata: int
+    commondataproc: str
+    nkin: int
+    nsys: int
+    commondata_table: pd.DataFrame = dataclasses.field(repr=False)
+    systype_table: pd.DataFrame = dataclasses.field(repr=False)
+    systematics_table: pd.DataFrame = dataclasses.field(init=None, repr=False)
+
+
+    def with_MULT_sys(self,mult_sys):
+        """
+        returns an InconsistentCommonData instance 
+        with MULT systematics replaced by mult_sys
+
+        Parameters
+        ----------
+        mult_sys : pd.DataFrame()
+                 all MULT columns of 
+                 InconsistentCommonData.commondata_table
+        """
+        table = self.commondata_table.copy()
+        table["MULT"] = mult_sys
+        return dataclasses.replace(self, commondata_table = table)
+    
+    def with_ADD_sys(self,add_sys):
+        """
+        returns an InconsistentCommonData instance
+        with ADD systematics replaced by add_sys
+
+        Parameters
+        ----------
+        add_sys : pd.DataFrame()
+                 all ADD columns of
+                 InconsistentCommonData.commondata_table
+        """
+        table = self.commondata_table.copy()
+        table["ADD"] = add_sys
+        return dataclasses.replace(self, commondata_table = table)
+
+    def rescale_sys(self,type_err,CORR,UNCORR,sys_rescaling_factor):
+        """
+        rescale the sys (MULT or ADD) by constant factor, sys_rescaling_factor,
+        a distinction is done between CORR and UNCORR systematics
+
+        Parameters
+        ----------
+
+        type_err : str 
+                e.g. 'MULT' or 'ADD'
+
+        CORR : bool
+
+        UNCORR : bool
+
+        sys_rescaling_factor : float, int
+
+        Returns
+        -------
+        pd.DataFrame corresponding to the rescaled MULT systematics
+        """
+        
+        err_table = self.systematics_table.loc[:,[type_err]].copy()
+        # get indices of CORR / UNCORR sys
+        systype_corr = self.systype_table[(self.systype_table["type"] == type_err) 
+                            & (~self.systype_table["name"].isin(["UNCORR","THEORYUNCORR"]))]
+        
+        systype_uncorr = self.systype_table[(self.systype_table["type"] == type_err) 
+                            & (self.systype_table["name"].isin(["UNCORR","THEORYUNCORR"]))]
+
+        # rescale systematics
+        if CORR:
+            err_table.iloc[:,systype_corr.index - 1] *= sys_rescaling_factor
+        if UNCORR:
```

Comment:
Since the actual braching is happening before, could you do move the `if CORR` and `if UNCORR` before (so when defining systype) and then just call `err_table.iloc[:,systype_corr.index - 1] *= sys_rescaling_factor` on the `systype` defined before?

Reply:
Docs added